packages/infrastructure-lib/src/mongo_settlement_transfer_repo.ts

```diff
@@ -100,10 +103,39 @@ export class MongoSettlementTransferRepo implements ISettlementBatchTransferRepo
 		}
 	}
 
-	async getBatchTransfersByBatchIds(batchIds: string[]): Promise<ISettlementBatchTransfer[]> {
+	async getBatchTransfersByBatchIds(
+		batchIds: string[],
+		pageIndex: number = 0,
+        pageSize: number = MAX_ENTRIES_PER_PAGE,
+	): Promise<BatchTransferSearchResults> {
 		try {
-			const batches = await this._collection.find({batchId: {$in: batchIds}}).project({_id: 0}).toArray();
-			return batches as ISettlementBatchTransfer[];
+			// Pagination settings
```

Comment:
In these methods, because we call them in many places in the aggragte, we need the default to be return all (without pagination)